### ISSUE TITLE
If there's any way to cancel the output archive file (.a) of cc_library ?

### ISSUE THREAD

**Issue by Okayu0010** [bazelbuild/bazel#20303]
I use cc_library rule to generate some middle object files, and merge these object files in some other places. It's unsafe and Inefficient if these middle object files be merged into a isolate archive file by cc_library. So I hope there's a way to cancel the output archive file (.a) of a cc_library target.

For example, I have a.cpp and b.cpp:
```
cc_library(
    name = "a",
    srcs = ["a.cpp"]
)
cc_library(
    name = "b",
    srcs = ["b.cpp"],
    deps = [":a"]
)
```

The libb.a file is needed and the liba.a should not be generated. I wonder if there's any to avoid generating the liba.a file?

**Comment by PikachuHyA** [comment]
I build with `bazel build :b`
```
INFO: Analyzed target //:b (36 packages loaded, 160 targets configured).
INFO: Found 1 target...
Target //:b up-to-date:
  bazel-bin/libb.a
  bazel-bin/libb.so
INFO: Elapsed time: 2.436s, Critical Path: 0.21s
INFO: 7 processes: 3 internal, 4 linux-sandbox.
INFO: Build completed successfully, 7 total actions
```
the `liba.a` is not generated

```
#tree bazel-bin/
bazel-bin/
├── libb.a
├── libb.a-2.params
├── libb.so
├── libb.so-2.params
└── _objs
    ├── a
    │   ├── a.pic.d
    │   └── a.pic.o
    └── b
        ├── b.pic.d
        └── b.pic.o

3 directories, 8 files
```

**Comment by comius** [comment]
Archives are usually not generated when library is used as a dependency. You get them if you build the library directly.

**Comment by Okayu0010** [comment]
Thanks for your effort. I have solved this problem now.
When I use cc_library and name it [foo], the libfoo.a file is generated as a default output. 
If the target [foo] is depended by a cc_binary target named [bar], them libbar.so link libfoo.a automatically that causes the libfoo.a be generated before all. However if [foo] is depended by a cc_library target named [baz], the libfoo.a won't be generated for libbaz.a doesn't demand it.
To solve my problem, I write a customized wrapper rule named my_cc_library, which replace the default output to *.o files instead of .a file. In addition, I avoid my_cc_library being depended by a cc_binary rule.
Closing this issue.
